YatimaStdLib/NonEmpty.lean

```diff
@@ -14,26 +14,20 @@ import YatimaStdLib.Traversable
 import YatimaStdLib.List
 
 structure NEList (α : Type u) where
-  data : List α
-  ne   : data ≠ []
+  head : α
+  tail : List α
 
 instance [ToString α] : ToString (NEList α) where
-  toString xs := toString xs.data
+  toString xs := toString (xs.head :: xs.tail)
 
 namespace NEList
 
 instance : CoeDep (List α) (x :: xs) (NEList α) where
```

Comment:
I think we should keep this; it will make the `List`/`NEList` pattern matching conincide. Just define
```
instance : CoeDep (List α) (x :: xs) (NEList α) where
  coe := { head := x, tail := xs }
```